samples/extensions/fragment_shader_barycentric/fragment_shader_barycentric.h

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2023, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+class FragmentShaderBarycentric : public ApiVulkanSample
+{
+  public:
+	struct
+	{
+		Texture envmap;
+		Texture cube;
+	} textures;
+
+	struct UBOVS
+	{
+		glm::mat4 projection;
+		glm::mat4 modelview;
+	} ubo_vs;
+
+	struct GUI_settings
+	{
+		int                            selected_effect = 0; /* default interpolation */
+		const std::vector<std::string> effect_names    = {"COLOR_INTERP",
+		                                                  "DIFF_PERSP_NOPERSP",
+		                                                  "WIREFRAME",
+		                                                  "INTERP_TO_MASS_CENTER",
+		                                                  "BARYCOORD_TEXTURE"};
```

Comment:
As these are strings in the UI, you might use some text that is better human-readable? Like "color interpolation", "perspective vs. non-perspective",...